### PR TITLE
NCGenerics: it's not experimental

### PR DESCRIPTION
**Pull request by kavon**
This PR promotes the experimental feature to a full language feature. Also fixes a bug where you could use the feature without the right flags in the swiftinterface file for feature-guarded declarations (rdar://127701059)

### CODE REVIEW

**Comment by kavon** [comment]
@swift-ci test

File: include/swift/Basic/Features.def [hunk 1]
```diff
@@ -179,6 +179,7 @@ LANGUAGE_FEATURE(MoveOnlyPartialConsumption, 429, "Partial consumption of noncop
 /// Enable bitwise-copyable feature.
 LANGUAGE_FEATURE(BitwiseCopyable, 426, "BitwiseCopyable protocol")
 SUPPRESSIBLE_LANGUAGE_FEATURE(ConformanceSuppression, 426, "Suppressible inferred conformances")
+SUPPRESSIBLE_LANGUAGE_FEATURE(NoncopyableGenerics, 426, "Noncopyable generics")
```
Comment (kavon) [inline include/swift/Basic/Features.def:182]: it's 427

**Comment by kavon** [comment]
@swift-ci test

**Review by egorzhdan — APPROVED** [review]
The C++ interop bits LGTM!

**Pull request merged**